### DISCUSSION TITLE
Set symbol to center of line

### DISCUSSION THREAD

**Discussion by 4tler** [mapsforge/vtm#866] [Q&A] [answered]
Is there a way to position a symbol in the middle of a line? So that it looks like it is a point element like a store.

I played around with a few options but the results were not satisfactory. 

results with symbol:
```
  <m v="put_in|egress|put_in;egress" closed="no">
      <symbol src="file:/symbols/whitewater_putin_egress.svg" symbol-height="25"/>
  </m>
```
![image](https://user-images.githubusercontent.com/74738662/128092519-ca4177e5-07b6-4ee7-ab8e-eff2efc9b88c.png)

results with lineSymbol:
```
  <m v="put_in|egress|put_in;egress" closed="no">
      <lineSymbol src="file:/symbols/whitewater_putin_egress.svg" symbol-height="25"/>
  </m>
```
![image](https://user-images.githubusercontent.com/74738662/128092888-36c07c11-ab84-4f24-9816-a3a238a5a74a.png)

I have overlooked something?

**Comment by devemux86** [comment, marked as answer]
Lines do not have centers, lines always split into multiple segments.

Line symbols are repeated along the path of the way elements.

If you want symbols in fixed positions, you should use node elements.

**Reply by 4tler** [reply]
Thanks for the quick reply. 

If I understand correctly, it means that I have to interprocess the osm data to get a center point from each line.
Can I formulate this as a kind of feature request? It would be something like the label-position="true" tag for polygons.

**Comment by devemux86** [comment]
Polygons are different from lines, the map can have lakes or buildings as polygons with labels / symbols at centroid.
However, even this cannot be perfect because of the map tiles, so a node label / symbol is always the best solution.

> it means that I have to interprocess the osm data to get a center point from each line.

This is the correct method: pre-process the osm before the map-writer.
As the lines are always divided into many segments, so there is no single center.

Lines are intended to behave like lines, with repeating labels / symbols throughout the line geometry.

**Reply by 4tler** [reply]
ok, thanks ...
